src/component/feature/FeatureView2/FeatureOverview/FeatureOverviewEnvSwitches/FeatureOverviewEnvSwitch/FeatureOverviewEnvSwitch.tsx

```diff
@@ -0,0 +1,92 @@
+import useFeature from '../../../../../../hooks/api/getters/useFeature/useFeature';
+import { TSetToastData } from '../../../../../../hooks/useToast';
+import { IFeatureEnvironment } from '../../../../../../interfaces/featureToggle';
+import { IFeatureViewParams } from '../../../../../../interfaces/params';
+import PermissionSwitch from '../../../../../common/PermissionSwitch/PermissionSwitch';
+import StringTruncator from '../../../../../common/StringTruncator/StringTruncator';
+import { UPDATE_FEATURE } from '../../../../../providers/AccessProvider/permissions';
+
+interface IFeatureOverviewEnvSwitchProps {
+    env: IFeatureEnvironment;
+    setToastData: TSetToastData;
+    callback?: () => void;
+}
+
+const FeatureOverviewEnvSwitch = ({
+    env,
+    setToastData,
+    callback,
+}: IFeatureOverviewEnvSwitchProps) => {
+    const { featureId, projectId } = useParams<IFeatureViewParams>();
+    const { toggleFeatureEnvironmentOn, toggleFeatureEnvironmentOff } =
+        useFeatureApi();
+    const { refetch } = useFeature(projectId, featureId);
+
+    const handleToggleEnvironmentOn = async () => {
+        try {
+            await toggleFeatureEnvironmentOn(projectId, featureId, env.name);
+            setToastData({
+                type: 'success',
+                show: true,
+                text: 'Successfully turned environment on.',
+            });
+            refetch();
+            if (callback) {
+                callback();
+            }
+        } catch (e) {
+            setToastData({
+                show: true,
+                type: 'error',
+                text: e.toString(),
+            });
+        }
+    };
+
+    const handleToggleEnvironmentOff = async () => {
+        try {
+            await toggleFeatureEnvironmentOff(projectId, featureId, env.name);
+            setToastData({
+                type: 'success',
+                show: true,
+                text: 'Successfully turned environment off.',
+            });
+            refetch();
+            if (callback) {
+                callback();
+            }
+        } catch (e) {
+            setToastData({
+                show: true,
+                type: 'error',
+                text: e.toString(),
+            });
+        }
+    };
+
+    const toggleEnvironment = async (e: React.ChangeEvent) => {
+        if (env.enabled) {
+            await handleToggleEnvironmentOff();
+            return;
+        }
+        await handleToggleEnvironmentOn();
+    };
+
+    return (
+        <div style={{ display: 'flex', alignItems: 'center' }}>
+            <PermissionSwitch
+                permission={UPDATE_FEATURE}
+                projectId={projectId}
+                checked={env.enabled}
+                onChange={toggleEnvironment}
+                tooltip={''}
+            />
+            <StringTruncator text={env.name} maxWidth="120" />
+            <span data-loading>is {env.enabled ? 'enabled' : 'disabled'}</span>
+        </div>
```

Comment:
We need to remove the extra white space after the environment name.
<img width="867" alt="Screen Shot 2021-11-10 at 9 23 50 PM" src="https://user-images.githubusercontent.com/47088512/141189142-5658c2cf-2fa5-4399-9353-84108dad005c.png">

Reply:
Yeah. This is actually due to the StringTruncator which needs a fixed width to add the ellipsis if the string is too long (See the really long environment name for example). I'm not quite sure yet how to handle this, but I agree that something needs to be done.

Reply:
We can switch the other `enabled in default          `

Reply:
I resolved it with maxWidth actually. I was wrong in terms of needing the fixed width :)